### PR TITLE
Adding some searching

### PR DESCRIPTION
**Pull request by bipol**
Playing around w/ swift -- wanted to try to implement the search bar here!

Not sure if you already have done this locally, but nonetheless was a fun experiment.
Simulator Screen Shot - iPhone 11 Pro Max - 2020-02-13 at 20 30 25
Simulator Screen Shot - iPhone 11 Pro Max - 2020-02-13 at 20 30 28

### CODE REVIEW

**Comment by glsmkr** [comment]
Oooooh, someone has been hiding their knowledge of iOS! 😂

File: smarta/smarta/StationLinesTableViewController.swift [hunk 1]
```diff
@@ -108,6 +111,20 @@ class StationLinesTableViewController: UITableViewController {
         }
     }
     
+    func searchBar(_ searchBar: UISearchBar, textDidChange searchText: String) {
+        if searchText.count == 0 {
```
Comment (glsmkr) [inline smarta/smarta/StationLinesTableViewController.swift]: I think you should be able to use this instead here.

```
        if searchText.isEmpty {
```

File: smarta/smarta/StationLinesTableViewController.swift [hunk 2]
```diff
@@ -108,6 +111,20 @@ class StationLinesTableViewController: UITableViewController {
...
+            displayedStations = stations
+        } else {
+            displayedStations = stations.filter({ (station) -> Bool in
+                let tmp: NSString = station.station.name as NSString
+                let range = tmp.range(of: searchText, options: NSString.CompareOptions.caseInsensitive)
+                return range.location != NSNotFound
+            })
+        }
+
+        self.tableView.reloadData()
```
Comment (glsmkr) [inline smarta/smarta/StationLinesTableViewController.swift]: This needs to happen on the main thread, so you wrap it like such.
```suggestion
DispatchQueue.main.async {
        self.tableView.reloadData()
        }
```

Comment (bipol) [reply smarta/smarta/StationLinesTableViewController.swift]: ahh roger -- joe brought up the main queue vs. other queues.  Will apply this change!

File: smarta/smarta/StationLinesTableViewController.swift (hunk 1 above)
Comment (bipol) [reply smarta/smarta/StationLinesTableViewController.swift]: ah perfect!

**Comment by glsmkr** [comment]
Looks good. Merge away!

**Review by glsmkr — APPROVED** [review]
WOOOH!

**Pull request merged**